satpy/readers/mviri_l1b_fiduceo_nc.py

```diff
@@ -558,13 +586,24 @@ def __init__(self, filename, filename_info, filetype_info,  # noqa: D417
             chunks={"x": CHUNK_SIZE,
                     "y": CHUNK_SIZE,
                     "x_ir_wv": CHUNK_SIZE,
-                    "y_ir_wv": CHUNK_SIZE}
+                    "y_ir_wv": CHUNK_SIZE},
+            decode_cf=False,
+            decode_times=False,
+            mask_and_scale=False,
         )
+
         self.nc = DatasetWrapper(nc_raw)
+        # decode times and fix duplicate dimensions
+        self.nc.prepare_input()
```

Comment:
I'd prefer this to be done in `DatasetWrapper.__init__`

Reply:
Thanks for reviewing!
I moved it back into DatasetWrapper.__init__ and adapted test_fix_duplicate_dimensions().
However, test_reassign_coords() now fails as it expects data to be time decoded - this was the original reason why I put it into a separate function. It should be possible to adapt the test, but the test case would get much bigger and test everything in DatasetWrapper.__init__ instead of only the dimension reassignment.
Any better idea is welcome ;)